### PR TITLE
[diag] use OT_ARRAY_LENGTH to unify code and shrink image size

### PR DESCRIPTION
**Pull request by yuzhyang**
nordic ot-cli-ftd hex: 38874 -> 3886c

### CODE REVIEW

File: third_party/mbedtls/repo/programs/test/selftest.c [hunk 1]
```diff
@@ -343,19 +342,19 @@ int main( int argc, char *argv[] )
         /* Run the specified tests */
         for( ; *argp != NULL; argp++ )
         {
-            for( test = selftests; test->name != NULL; test++ )
+            for( size_t i = 0; i < OT_ARRAY_LENGTH(selftests); i++ )
```
Comment (LuDuda) [inline third_party/mbedtls/repo/programs/test/selftest.c]: I would not modify the original `mbedtls` test. What is the reason behind it?

File: src/core/common/code_utils.hpp [hunk 2]
```diff
@@ -37,7 +37,6 @@
 #include "openthread-core-config.h"
 
 #include "utils/static_assert.hpp"
```
Comment (bukepo) [inline src/core/common/code_utils.hpp]: delete this line too?

File: examples/platforms/nrf52840/diag.c [hunk 3]
```diff
@@ -45,6 +45,7 @@
 #include <openthread/platform/radio.h>
 #include <openthread/platform/toolchain.h>
 
+#include <common/code_utils.hpp>
```
Comment (abtink) [inline examples/platforms/nrf52840/diag.c]: The  `common/code_utils.hpp` is intended as internal to OT core header file (with C++ definitions which may not work in a C-99 source).  We have `platforms/utils/code_utils.h` for platform code to use. Please check this PR https://github.com/openthread/openthread/pull/1534.

File: src/core/common/code_utils.hpp [hunk 4]
```diff
@@ -37,7 +37,6 @@
...
-#include "utils/wrap_stdbool.h"
```
Comment (abtink) [inline src/core/common/code_utils.hpp]: Any reason for removing `stdbool.h`? 

I think we need it for definition of `false` (and `true`) which is used in the macros `VerifyOrExit`, 'SuccessOrExit`

File: examples/platforms/nrf52840/diag.c (hunk 3 above)
Comment (yuzhyang) [reply examples/platforms/nrf52840/diag.c]: Thanks for the kind explanations. Will modify accordingly.

File: src/core/common/code_utils.hpp (hunk 4 above)
Comment (yuzhyang) [reply src/core/common/code_utils.hpp]: Have tried locally. Will add it back

**Pull request merged**